src/fsharp/LexFilter.fs

```diff
@@ -677,105 +704,119 @@ type LexFilterImpl (lightStatus: LightSyntaxStatus, compilingFsLib, lexer, lexbu
     // Undentation rules
     //--------------------------------------------------------------------------
 
+    let (|RelaxWhitespace2|_|) x = if lexbuf.SupportsFeature LanguageFeature.RelaxWhitespace2 then Some x else None
     let pushCtxt tokenTup (newCtxt: Context) =
         let rec undentationLimit strict stack = 
             match newCtxt, stack with 
             | _, [] -> PositionWithColumn(newCtxt.StartPos, -1) 
 
             // ignore Vanilla because a SeqBlock is always coming 
             | _, (CtxtVanilla _ :: rest) -> undentationLimit strict rest
-
-            | _, (CtxtSeqBlock _ :: rest) when not strict -> undentationLimit strict rest
-            | _, (CtxtParen _ :: rest) when not strict -> undentationLimit strict rest
+            | (_, (CtxtSeqBlock _ :: rest) | _, (CtxtParen _ :: rest)) when not strict
+                -> undentationLimit strict rest
 
             // 'begin match' limited by minimum of two  
             // '(match' limited by minimum of two  
-            | _, (((CtxtMatch _) as ctxt1) :: CtxtSeqBlock _ :: (CtxtParen ((BEGIN | LPAREN), _) as ctxt2) :: _rest)
-                      -> if ctxt1.StartCol <= ctxt2.StartCol 
-                         then PositionWithColumn(ctxt1.StartPos, ctxt1.StartCol) 
-                         else PositionWithColumn(ctxt2.StartPos, ctxt2.StartCol) 
-
-             // 'let ... = function' limited by 'let', precisely  
-             // This covers the common form 
-             //                          
-             //     let f x = function   
-             //     | Case1 -> ...       
-             //     | Case2 -> ...       
-            | (CtxtMatchClauses _), (CtxtFunction _ :: CtxtSeqBlock _ :: (CtxtLetDecl _ as limitCtxt) :: _rest)
-                      -> PositionWithColumn(limitCtxt.StartPos, limitCtxt.StartCol)
-
-            // Otherwise 'function ...' places no limit until we hit a CtxtLetDecl etc... (Recursive) 
-            | (CtxtMatchClauses _), (CtxtFunction _ :: rest)
-                      -> undentationLimit false rest
-
-            // 'try ... with' limited by 'try'  
-            | _, (CtxtMatchClauses _ :: (CtxtTry _ as limitCtxt) :: _rest)
-                      -> PositionWithColumn(limitCtxt.StartPos, limitCtxt.StartCol)
-
-            // 'fun ->' places no limit until we hit a CtxtLetDecl etc... (Recursive) 
-            | _, (CtxtFun _ :: rest)
-                      -> undentationLimit false rest
+            | _, ((CtxtMatch _ as ctxt1) :: CtxtSeqBlock _ :: (CtxtParen ((BEGIN | LPAREN), _) as ctxt2) :: _rest)
```

Comment:
Please remove any changes that are just cleanup.  To assess this we need a PR with an absoltuely minimal diff